tortoise/ninja_tortoise.go

```diff
@@ -490,35 +499,21 @@ func (ni *NinjaTortoise) updatePatSupport(p votingPattern, bids []types.BlockID,
 	ni.TPatSupport[p][idx] = votingPattern{id: pid, LayerID: idx}
 }
 
-func initTallyToBase(tally map[votingPattern]map[types.BlockID]vec, base votingPattern, p votingPattern) {
-	tally[p] = make(map[types.BlockID]vec)
-	for k, v := range tally[base] {
-		tally[p][k] = v
+func (ni *NinjaTortoise) initTallyToBase(base votingPattern, p votingPattern, windowStart types.LayerID) {
+	mp := make(map[BlockIDLayerTuple]vec, len(ni.TTally[base]))
```

Comment:
Out of curiosity, why do you define a new map `mp`, and not directly `ni.TTally[p]` as above?

Reply:
less callsnto ttally